src/dbnode/persist/fs/seek_manager.go

```diff
@@ -108,11 +111,35 @@ type borrowableSeeker struct {
 	isBorrowed bool
 }
 
-type seekersByTime struct {
+type seekersByTimeAndVolume struct {
 	sync.RWMutex
 	shard    uint32
 	accessed bool
-	seekers  map[xtime.UnixNano]seekersAndBloom
+	seekers  map[xtime.UnixNano]map[int]seekersAndBloom
```

Comment:
I dont think volume should be parse of the map. There should only been seekers/bloom for one volume at a time (you may need to keep track of which volume you have but you shouldn't have multiple at once)